### PR TITLE
Initial project setup for peak finding algorithm.

### PR DESCRIPTION
**Pull request by Daroslaw**
A base for further development. I hope I did not forget about anything :P

### CODE REVIEW

File: src/native-algorithms.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.26730.12
+VisualStudioVersion = 15.0.27130.2027
```
Comment (Sand3r-) [inline src/native-algorithms.sln]: Awaiting "unnecessary change" from @gmrukwa

**Review by Sand3r- — DISMISSED** [review]

File: src/Spectre.libPeakFinder.Tests/Main.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
```
Comment (gmrukwa) [inline src/Spectre.libPeakFinder.Tests/Main.cpp]: I've been usually just pointing all the test projects to existing `Common\Main.cpp` file, so the same file is in fact compiled in each of test projects ;)

File: src/Spectre.libPeakFinder.Tests/Spectre.libPeakFinder.Tests.vcxproj [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+    <Import Project="..\PropertySheets\NativeTestProject.props" />
+  </ImportGroup>
+  <PropertyGroup Label="UserMacros" />
+  <PropertyGroup />
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+    <ClCompile>
+      <WarningLevel>Level4</WarningLevel>
```
Comment (gmrukwa) [inline src/Spectre.libPeakFinder.Tests/Spectre.libPeakFinder.Tests.vcxproj:79]: `TreatWarningsAsErrors` is missing.

File: src/Spectre.libPeakFinder/Spectre.libPeakFinder.vcxproj [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+    <Import Project="..\PropertySheets\NativeProject.props" />
+  </ImportGroup>
+  <PropertyGroup Label="UserMacros" />
+  <PropertyGroup />
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+    <ClCompile>
+      <WarningLevel>Level4</WarningLevel>
```
Comment (gmrukwa) [inline src/Spectre.libPeakFinder/Spectre.libPeakFinder.vcxproj:79]: `TreatWarningsAsErrors` is missing.

File: src/Spectre.libPeakFinder.Tests/Spectre.libPeakFinder.Tests.vcxproj (hunk 3 above)
Comment (Daroslaw) [reply src/Spectre.libPeakFinder.Tests/Spectre.libPeakFinder.Tests.vcxproj:79]: image
Goddamnit

**Review by Sand3r- — DISMISSED** [review]

**Review by Sand3r- — APPROVED** [review]
